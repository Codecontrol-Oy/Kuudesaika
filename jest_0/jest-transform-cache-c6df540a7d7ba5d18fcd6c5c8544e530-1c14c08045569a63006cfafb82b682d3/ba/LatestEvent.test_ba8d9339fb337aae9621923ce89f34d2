'use strict';




jest.mock('react-router', function () {return { browserHistory: { push: jest.fn() } };});var _react = require('react');var _react2 = _interopRequireDefault(_react);var _LatestEvent = require('./LatestEvent');var _LatestEvent2 = _interopRequireDefault(_LatestEvent);var _Organization = require('../../../mock/Organization.json');var _Organization2 = _interopRequireDefault(_Organization);var _sinon = require('sinon');var _sinon2 = _interopRequireDefault(_sinon);var _reactRouter = require('react-router');function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe("<LatestEvent />", function () {

    it("Matches snapshot", function () {
        var latestEvent = shallow(
        _react2.default.createElement(_LatestEvent2.default, { organization: _Organization2.default }));

        expect(latestEvent).toMatchSnapshot();
    });

    it("Should redirect to organization on click", function () {

        var latestEvent = mount(
        _react2.default.createElement(_LatestEvent2.default, { organization: _Organization2.default }));

        var callback = _sinon2.default.spy(latestEvent.instance(), 'onLinkAction');
        var organizationLink = latestEvent.find('a');
        organizationLink.simulate('click');
        expect(callback.calledOnce).toEqual(true);
        expect(_reactRouter.browserHistory.push).toHaveBeenCalled();
    });
});