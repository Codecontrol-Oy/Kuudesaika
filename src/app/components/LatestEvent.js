import React from 'react';
import { connect } from 'react-redux';
import {browserHistory} from 'react-router';
import PropTypes from 'prop-types';
import Card from 'grommet/components/Card';
import Label from 'grommet/components/Label';
import Anchor from 'grommet/components/Anchor';
import Timestamp from 'grommet/components/Timestamp';
import axios from 'axios';
export default class LatestEvent extends React.Component {
  constructor (props) {
    super(props);
  }

  onLinkAction = () => {
    browserHistory.push("/organisaatio/" + this.props.organization_id);
  }

  render () {
    return (
        <div>
            <Card contentPad={"medium"} 
                  style={{borderBottom: "2px solid #CCC"}}
                  label={<Timestamp fields={"date"} value={this.props.start_date} />}
                  link={<Anchor onClick={() => this.onLinkAction()} >Siirry</Anchor>}
                  heading={this.props.organization} />
        </div>
    );
  }
}

LatestEvent.propTypes = {
    organization: PropTypes.string,
    start_date: PropTypes.string,
    organization_id: PropTypes.string,
    id: PropTypes.number
};