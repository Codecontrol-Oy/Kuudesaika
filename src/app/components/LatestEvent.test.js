import React from 'react';
import LatestEvent from './LatestEvent';
import organizationMock from '../../../mock/Organization.json';

describe("<LatestEvent />", () => {
    it("Matches snapshot", () => {
    const latestEvent = shallow(
        <LatestEvent organization={organizationMock}  />
    );
    expect(latestEvent).toMatchSnapshot();
    });
}) 
