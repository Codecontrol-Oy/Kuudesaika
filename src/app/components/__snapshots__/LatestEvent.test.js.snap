// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LatestEvent /> Matches snapshot 1`] = `
<div>
  <Card
    a11yTitle="Card"
    contentPad="medium"
    heading="Kaupunginhallitus"
    headingStrong={true}
    label="helsinki-Hallitus"
    link={
      <Anchor
        animateIcon={true}
        method="push"
        onClick={[Function]}
        tag="a"
      >
        Siirry
      </Anchor>
    }
    style={
      Object {
        "borderBottom": "2px solid #CCC",
      }
    }
    textSize="small"
  />
</div>
`;
